components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -4623,24 +4631,98 @@ public static boolean jsFunction_hasUserPermissions(Context cx, Scriptable thisO
+        String username = "";
+
         try {
-            APIManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
-            String url = config.getFirstProperty(APIConstants.AUTH_MANAGER_URL);
             if (url == null) {
                 handleException("API key manager URL unspecified");
             }
-            stub = new UserRegistrationAdminServiceStub(null, url + "UserRegistrationAdminService");
-            ServiceClient client = stub._getServiceClient();
-            Options option = client.getOptions();
+
+            AuthenticationAdminStub authAdminStub = new AuthenticationAdminStub(null, url + "AuthenticationAdmin");
+            ServiceClient client = authAdminStub._getServiceClient();
+            Options options = client.getOptions();
+            options.setManageSession(true);
+
+            boolean isTenantFlowStarted = false;
+
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            }
+            // get the signup configuration
+            UserRegistrationConfigDTO signupConfig = SelfSignUpUtil.getSignupConfiguration(tenantDomain);
+            // set tenant specific sign up user storage
+            if (signupConfig != null && !signupConfig.getSignUpDomain().isEmpty()) {
+                if (!signupConfig.isSignUpEnabled()) {
+                    handleException("Self sign up has been disabled for" + tenantDomain);
+                }
+            }
+            username = signupConfig.getAdminUserName();
+            String password = signupConfig.getAdminPassword();
+
+            String host = null;
+            host = new URL(url).getHost();
+            if (!authAdminStub.login(username, password, host)) {
+                handleException("Login failed for " + username + " Please recheck the username and password and try again.");
+            }
+
+            ServiceContext serviceContext = authAdminStub.
+                    _getServiceClient().getLastOperationContext().getServiceContext();
+            String sessionCookie = (String) serviceContext.getProperty(HTTPConstants.COOKIE_STRING);
+
+            stub = new ClaimMetadataManagementServiceStub(null, url + "ClaimMetadataManagementService");
+            ServiceClient client1 = stub._getServiceClient();
+            Options option = client1.getOptions();
             option.setManageSession(true);
-            userFields = stub.readUserFieldsForUserRegistration(UserCoreConstants.DEFAULT_CARBON_DIALECT);
+            option.setProperty(HTTPConstants.COOKIE_STRING, sessionCookie);
+
+            org.wso2.carbon.identity.claim.metadata.mgt.stub.dto.LocalClaimDTO[] localClaimDTOS = null;
```

Comment:
Why define full class name with package names, while the class is already imported?

Reply:
Fixed